lib/pulsar/broker.ex

```diff
@@ -701,6 +721,22 @@ defmodule Pulsar.Broker do
     end
   end
 
+  defp handle_command(%Binary.CommandProducerSuccess{} = command, broker) do
+    # CommandProducerSuccess can arrive twice for WaitForExclusive mode:
+    # 1. First with producer_ready: false (pending state) - request_id in broker.requests
+    # 2. Second with producer_ready: true (final state) - request_id NOT in broker.requests (already cleaned up)
+    request_id = command.request_id
+
+    # Check if this request_id is still pending in broker.requests
+    if Map.has_key?(broker.requests, request_id) do
+      # Initial registration response - request is still pending
+      handle_producer_registration_response(command, broker, request_id)
```

Comment:
I understand the additional logic may be to map producer names to producer ids. Depending on how we go about my first comment, it may be enough to just call `reply_to_request/3` as we do in other places?